lib/trino-hive-formats/src/main/java/io/trino/hive/formats/avro/model/EnumReadAction.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hive.formats.avro.model;
+
+import io.airlift.slice.Slice;
+import org.apache.avro.Schema;
+
+import static java.util.Objects.requireNonNull;
+
+public final class EnumReadAction
+        extends AvroReadAction
+{
+    private final Schema readSchema;
+    private final Schema writeSchema;
+    // Decoder::readInt returns the index of the symbol to use
+    private final Slice[] symbolIndex;
```

Comment:
I'd use a list here also.  Looking at the usage, I don't think it will have any impact on performance